runtime/compiler/env/j9methodServer.cpp

```diff
@@ -1236,6 +1301,11 @@ TR_ResolvedJ9JITaaSServerMethod::setAttributes(TR_OpaqueMethodBlock * aMethod, T
   
    setMandatoryRecognizedMethod(mandatoryRm);
    setRecognizedMethod(rm);
+
+   TR_JITaaSIProfiler *iProfiler = (TR_JITaaSIProfiler *) ((TR_J9VMBase *) fe)->getIProfiler();
+   const std::string entryStr = std::get<3>(methodInfo);
+   const auto serialEntry = (TR_ContiguousIPMethodHashTableEntry*) &entryStr[0];
+   _iProfilerMethodEntry = (TR::comp() && iProfiler && !entryStr.empty()) ? iProfiler->deserializeMethodEntry(serialEntry) : nullptr; 
```

Comment:
I think that here TR::comp() can return nullptr because when first create the `compilee` and then we create the compilation object. However, we don't want to have _iProfilerMethodEntry null for the method to be compiled. I think that the only reason you want the test for TR:comp() is because `deserializeMethodEntry` needs to allocate heap memory. In this case it could be easier if we provide it with a pointer to trMemory object available in this routine.